.github/workflows/security.yml

```
@@ -0,0 +1,90 @@
+    push:
+      branches: ["AG-3390-snyk-gha"]
+    pull_request:
+      branches: ["AG-3390-snyk-gha"]
+
+jobs:
+  security-all-projects:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'issue_comment' && github.event.comment.body == '/security') || github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@main
+      - name: Echo ignore lists
+        run: |
+          echo "SECURITY_BASE_IGNORE_LIST: ${{ env.SECURITY_BASE_IGNORE_LIST }}"
+          echo "SECURITY_PACKAGE_IGNORE_LIST: ${{ env.SECURITY_PACKAGE_IGNORE_LIST }}"
+      - name: Run snyk to check for vulnerabilities on all projects
+        uses: snyk/actions/node@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          args: --all-projects --severity-threshold=${{ env.SECURITY_SEVERITY_THRESHOLD }} --exclude=${{ env.SECURITY_BASE_IGNORE_LIST }},${{ env.SECURITY_PACKAGE_IGNORE_LIST }}
+
+  security-ignored-paths:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'issue_comment' && github.event.comment.body == '/security') || github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@main
+      - name: Echo ignore lists
+        run: |
+          echo "SECURITY_PACKAGE_IGNORE_LIST_PATHS: ${{ env.SECURITY_PACKAGE_IGNORE_LIST_PATHS }}"
+      - name: Run snyk test individually on each ignored path
+        shell: bash
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        run: |
+          IFS=',' read -ra PATHS <<< "${{ env.SECURITY_PACKAGE_IGNORE_LIST_PATHS }}"
+          for path in "${PATHS[@]}"; do
+            echo "Scanning $path"
+
+            dir=$(dirname "$path")
+            echo "Working in directory: $dir"
+
+            if [ ! -f "$dir/package-lock.json" ] && [ ! -d "$dir/node_modules" ]; then
+              echo "No package-lock.json or node_modules found, which is required by snyk test — running npm install in $dir"
+              (cd "$dir" && npm install --force --ignore-scripts)
+            fi
+
+            echo "-----------------------------------------------------"
+            echo "-- snyk test: $path"
+            echo "-----------------------------------------------------"
+            npx snyk test --severity-threshold=${{ env.SECURITY_SEVERITY_THRESHOLD }} --file="$path"
+            echo "-----------------------------------------------------"
+          done
+
+  code-security:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'issue_comment' && github.event.comment.body == '/security') || github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'pull_request'
```

To be removed when review is passed

```suggestion
    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'issue_comment' && github.event.comment.body == '/security') || github.event_name == 'schedule'
```